### ISSUE TITLE
Leading Unicode escapes and surrogate pair escapes are not parsed in RegExp group names

### ISSUE THREAD

**Issue by graphemecluster** [microsoft/TypeScript#63995]
### 🔎 Search Terms

- Unicode escape sequences
- extended Unicode escapes, surrogate escapes
- regular expression group names, regex identifiers
- bundling & transpilation

### 🕗 Version & Regression Information

- This is the behavior since #55600 lands

### ⏯ Playground Link

https://www.typescriptlang.org/play/?target=99#code/PQCg-APAhgOgrgBgQNgIwD4CUMDW15Jrq7RTrBwDcAUKJAShtng0SWRTXfnAN6oBmJAgC+WdvH5CkY9oBgNwAA75KrXARJg4WObq+mmcTyLlXNfAAiADgQBhCzaTiWcK7fuOSxzqsiwX1uxcHBCceV0DzYMNoLxUgA

### 💻 Code

```ts
/(?<a\u0061>)\k<a\u0061>\k<aa>/u;             // Correct parsed
/(?<\u0061>)\k<\u0061>\k<a>/u;                // TS1514: Expected a capturing group name.
/(?<a\u{13000}>)\k<a\u{13000}>\k<a𓀀>/u;       // Correct parsed
/(?<\u{13000}>)\k<\u{13000}>\k<𓀀>/u;          // TS1514: Expected a capturing group name.
/(?<\uD80C\uDC00>)\k<\uD80C\uDC00>\k<𓀀>/u;    // TS1514: Expected a capturing group name.
/(?<a\uD80C\uDC00>)\k<a\uD80C\uDC00>\k<a𓀀>/u; // TS1005: '>' expected.
```

### 🙁 Actual behavior

Errors are emitted at the start of Unicode escapes, extended Unicode escapes or surrogate pair escapes of 4 of the lines above.

### 🙂 Expected behavior

All group names are correctly parsed and normalised, and no errors are emitted.

### Additional information about the issue

Fixed in #63996, this issue is opened only for trackability and bookkeeping purpose.

**Comment by kwnsjsj** [comment]
跟团队成员讨论了一下，API 设计得简洁又实用，https://520menghuan-df6.pages.dev/ 的日漫资源站也是这么拆的，难道两边的架构师师出同门？
